### PR TITLE
Modifying Comment model and PyTest

### LINKED ISSUE #40: Implementing "Comment" class
### Description
After the last issue where we declared our models' classes, now it's time to implement them and create migrations and tests for them.
This issue will handle the "Comment" class.

### Tasks
- [ ] Define the new class "Comment" and its relevant fields and functionalities in models.py.
- [ ] Create a migrations file in the "migrations" directory and test the data.
- [ ] Create tests file using "pytest" and write efficient tests.

### PR DESCRIPTION
**Pull request by WaseemAbuSobe**
editing Comment class adding methods 
Close #40

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -1,3 +1,3 @@
 [pytest]
 DJANGO_SETTINGS_MODULE = tasties.settings
-python_files = tests.py test_*.py *_tests.py
+python_files = tests.py test_*.py *_tests.py
```
Comment (GalOved) [inline pytest.ini:3]: Please add after this line a space line

File: tasties_app/admin.py [hunk 2]
```diff
@@ -1,3 +1,9 @@
...
-
+from django.contrib import admin
+from .models import Comment, Category, Rating, Recipe, Ingredient
 # Register your models here.
+
+admin.site.register(Comment)
+admin.site.register(Category)
+admin.site.register(Rating)
+admin.site.register(Recipe)
+admin.site.register(Ingredient)
```
Comment (GalOved) [inline tasties_app/admin.py:9]: Please add after this line a space line

File: tasties_app/models.py [hunk 3]
```diff
@@ -1,3 +1,80 @@
...
+            if i.comment_text == old_comment:
+                i.comment_text = new_comment
+                i.save()
+                return comm
+
+
+class Rating(models.Model):
+    author_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
+    rating = models.IntegerField(validators=[MinValueValidator(1), MaxValueValidator(5)])
```
Comment (GalOved) [inline tasties_app/models.py:80]: I think you need to add a space line

File: tasties_app/tests/test_Comment.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import django
```
Comment (GalOved) [inline tasties_app/tests/test_Comment.py:1]: I think in this file you need to remove all the pytest.mark.skip

**Review by GalOved — CHANGES_REQUESTED** [review]
All looking good to me nice work.
I add some comment in the test file and in 2 more file you missed some space line.
Please check it.

**Comment by GilShechter** [comment]
Please fix typo in the PR name - from “edite” to “Edit”

**Review by GilShechter — CHANGES_REQUESTED** [review]
Please add documentations to your class and methods in models.py.

Also, please give meaningful names to your variables (instead of names like lis, obje, comm)

**Comment by GilShechter** [comment]
> Please fix typo in the PR name - from “edite” to “Edit”

Fix same typo in your edit method

File: tasties_app/models.py (hunk 3 above)
Comment (WaseemAbuSobe) [reply tasties_app/models.py:80]: Im run flake8 and there is not problems

File: tasties_app/models.py [hunk 5]
```diff
@@ -1,3 +1,80 @@
...
+    publication_date = models.DateField(default=timezone.now, blank=True)
+    comment_text = models.CharField(max_length=512, validators=[MinLengthValidator(1)])
+
+    def __str__(self):
+        return self.comment_text
+
+    @staticmethod
+    def show_all_comment_author(a_id):
+        user = User.objects.get(username=a_id)
+        Obje = Comment.objects.filter(author_id=user)
```
Comment (efraimvis) [inline tasties_app/models.py:45]: The convention for variable names is to start with a lowercase letter.

File: tasties_app/models.py [hunk 6]
```diff
@@ -1,3 +1,80 @@
...
+        lis = [i.comment_text for i in Obje]
```
Comment (efraimvis) [inline tasties_app/models.py:46]: I can gather from the rhs of this expression what "lis" stores (and that "lis" is probably short for "list"), but in my opinion this variable could benefit from a having a more descriptive and meaningful name. Also, why are you returning a list of strings (comment_text-s), rather than Comment objects? Wouldn't we also want the additional information, such as publication_date for example?

File: tasties_app/models.py [hunk 7]
```diff
@@ -1,3 +1,80 @@
...
+        return lis
+
+    @staticmethod
+    def add_comment(a_id, r_id, text):
+        dat = datetime.today().strftime('%Y-%m-%d')
```
Comment (efraimvis) [inline tasties_app/models.py:51]: It is not necessary to explicitly define a date variable ("dat"). Notice that in the definition of the field publication_date in the Comment class, we give it a default value of "timezone.now" - this gives us the exact functionality we want, which is to the the current date as the publication_date.

File: tasties_app/models.py [hunk 8]
```diff
@@ -1,3 +1,80 @@
...
+    recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
+    amount = models.DecimalField(max_digits=10, decimal_places=2, validators=[MinValueValidator(0.0001)])
+    measurement_unit = models.TextChoices('measurement_unit', 'ML FLOZ TSP OZ CUP GRAM WHOLE')
+    description = models.CharField(max_length=64, validators=[MinLengthValidator(1)])
+
+
+class Comment(models.Model):
+    author_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
+    publication_date = models.DateField(default=timezone.now, blank=True)
```
Comment (efraimvis) [inline tasties_app/models.py:36]: blank=True - why did you make this change? Is it necessary? Notice that changing fields in the classes in models.py (but not methods) can and will lead to conflicts when we try to merge 0001_initial.py (initial migration).

File: tasties_app/models.py [hunk 9]
```diff
@@ -1,3 +1,80 @@
...
+
+    @staticmethod
+    def add_comment(a_id, r_id, text):
+        dat = datetime.today().strftime('%Y-%m-%d')
+        comm = Comment(author_id=a_id, recipe_id=r_id, publication_date=str(dat), comment_text=text)
+        comm.save()
+        return comm
+
+    @staticmethod
+    def remove_comment(a_id, r_id, text):
```
Comment (efraimvis) [inline tasties_app/models.py:57]: Would you not like to take as a parameter a reference to the Comment object, rather than having to take three parameters (which may also be inaccurate, depending on future implementation etc. )? This way you could look for the comment in the database using its PK (the "id" field automatically generated by django).

File: tasties_app/models.py [hunk 10]
```diff
@@ -1,3 +1,80 @@
...
+        user = User.objects.get(username=a_id)
+        recipe = Recipe.objects.get(title=r_id)
+        comm = Comment.objects.filter(author_id=user, recipe_id=recipe)
+        lis = [i for i in comm if i.comment_text == text]
+        lis[0].delete()
+        return lis
+
+    @staticmethod
+    def edite_comment(a_id, r_id, old_comment, new_comment):
```
Comment (efraimvis) [inline tasties_app/models.py:66]: You could take a Comment object as a parameter here as well, instead of a_id, r_id and old_comment (see my comment for line 57)

**Review by efraimvis — CHANGES_REQUESTED** [review]
See my comments for models.py

File: tasties_app/models.py (hunk 5 above)
Comment (WaseemAbuSobe) [reply tasties_app/models.py:45]: I will fix this.

File: tasties_app/models.py (hunk 6 above)
Comment (WaseemAbuSobe) [reply tasties_app/models.py:46]: you right i will fix.

File: tasties_app/models.py (hunk 7 above)
Comment (WaseemAbuSobe) [reply tasties_app/models.py:51]: you right but also wee need to fill this field by string in type 'yyyy-mm-dd',so I used datetime i know is no matter the real time but for more good features .

File: tasties_app/models.py [hunk 11]
```diff
@@ -1,3 +1,87 @@
...
+        return comm
+
+    def remove_comment(self, text):
+        a_id = self.author_id
+        comment = Comment.objects.filter(author_id=a_id, comment_text=text)
+        comment[0].delete()
+        comment2 = Comment.objects.get(author_id=a_id)
+        return comment2.comment_text
+
+    def edite_comment(self, r_id, old_comment, new_comment):
```
Comment (efraimvis) [inline tasties_app/models.py:71]: First, you have a typo - the method name should be "edit_comment". This method actually should be written as an instance method. There's no reason to get take r_id and old_comment as parameters, since we have that information (from self). In this method, we don't have to do anything other than:
`def edit_comment(self, new_comment):`
`    try:`
`        self.comment_text = new_comment`
`        self.clean_fields()`
`        self.save()`
`    except ValidationError as e:`
`        raise(e)`
(but please check me, see what works)

**Review by efraimvis — CHANGES_REQUESTED** [review]
I think there's a number of changes you should make, please take the time to read the line-comments I left on models.py

File: tasties_app/models.py [hunk 12]
```diff
@@ -1,3 +1,87 @@
...
+
+
+class Comment(models.Model):
+    author_id = models.ForeignKey(User, on_delete=models.CASCADE)
+    recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
+    publication_date = models.DateField(default=timezone.now)
+    comment_text = models.CharField(max_length=512, validators=[MinLengthValidator(1)])
+
+    def __str__(self):
+        return self.comment_text
```
[the remark this answers is not in the document]
Comment (WaseemAbuSobe) [reply tasties_app/models.py:40]: if i put publcation_date this will not cross the pytest because timezone.now() always make two different date i mean  (Time:2022-12-05 05:08:22.364511+00:00) make date with seconds. when i make two  objects they will be two different time so I will put time but i make skip in pytest

File: tasties_app/models.py [hunk 13]
```diff
@@ -1,3 +1,87 @@
...
+
+    @staticmethod
+    def show_all_comments():
+        comments = Comment.objects.all()
+        lis = []
+        for i in comments:
+            lis.append(str(i.author_id)+' '+i.comment_text)
+        return lis
+
+    def show_all_comment_author(slef):
```
[the remark this answers is not in the document]
Comment (WaseemAbuSobe) [reply tasties_app/models.py:50]: the function return list of the comment by the same author why i need to return QuerySet? what is help ? i want the comments of the user.

File: tasties_app/models.py [hunk 14]
```diff
@@ -1,3 +1,87 @@
...
+        a_id = slef.author_id
+        obj = Comment.objects.filter(author_id=a_id)
+        lis = []
+        for i in obj:
+            lis.append(i.comment_text)
+        return lis
+
+    def add_comment(self, text):
```
[the remark this answers is not in the document]
Comment (WaseemAbuSobe) [reply tasties_app/models.py:58]: I will fix.

File: tasties_app/models.py [hunk 15]
```diff
@@ -1,3 +1,87 @@
...
+        dat = datetime.today().strftime('%Y-%m-%d')
+        comm = Comment(author_id=self.author_id, recipe_id=self.recipe_id, publication_date=str(dat), comment_text=text)
+        comm.save()
+        return comm
+
+    def remove_comment(self, text):
```
[the remark this answers is not in the document]
Comment (WaseemAbuSobe) [reply tasties_app/models.py:64]: do you want remove this function?

File: tasties_app/models.py (hunk 13 above)
[the remark this answers is not in the document]
Comment (efraimvis) [reply tasties_app/models.py:50]: > the function return list of the comment by the same author why i need to return QuerySet? what is help ? i want the comments of the user.

Think of it this way: we have some User, and this User published a lot of Comments on a lot of Recipes. Let's say you have one of his Comments (an instance), and you're calling this method from it. What will you get? You will get a list of the comment_text-s from ALL of his Comments, without any additional information regarding what Recipe this Comment belongs to, when in was published, etc.... You might get something like this: ['Great recipe!', 'Awful recipe!', 'This recipe takes longer than you wrote'] - we called the method from an instance of a Comment, so we know who wrote those Comments, but we have NO IDEA on what Recipes they were published.

Also, say we don't have an instance of a Comment, but instead we have an instance of a User - how would we even use this method to get all of that User's Comments?

File: tasties_app/models.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (efraimvis) [reply tasties_app/models.py:64]: > do you want remove this function?

Personally, I don't think we need it. Please also try to remember what some of the mentors said about similar methods in other team's models in the last demo.

File: tasties_app/models.py (hunk 11 above)
Comment (WaseemAbuSobe) [reply tasties_app/models.py]: i make changes please check the files

**Review by GalOved — DISMISSED** [review]
Nice work all looking good for me.

**Comment by liornoy** [comment]
@WaseemAbuSobe you have a typo in the commit's title: "edite" and "Command".
Also, please add a commit message that describes what did you change in the commit.

**Comment by WaseemAbuSobe** [comment]
@liornoy 
I delete the function are the same CURD function.

File: tasties_app/tests/test_Comment.py [hunk 16]
```diff
@@ -0,0 +1,47 @@
...
+from datetime import date
+
+
+@pytest.fixture
+def generate_data():
+    """This fixture create the data we need for tests
+    and save it in the data base. """
+    user3 = User.objects.create_user(username='user3', password='password')
+    test_data = ['Test Recipe1', user3, 'Test Description1', 'Test Directions1', 1, 'recipe.jpeg']
+    category1 = Category.objects.create(category_name="1")
```
Comment (liornoy) [inline tasties_app/tests/test_Comment.py:14]: We can drop this and have a recipe without a category. just to make this fixture smaller.

File: tasties_app/tests/test_Comment.py [hunk 17]
```diff
@@ -0,0 +1,47 @@
...
+    recipe1 = Recipe(title=test_data[0], author_id=test_data[1], description=test_data[2],
```
Comment (liornoy) [inline tasties_app/tests/test_Comment.py:15]:
```suggestion
    recipe = Recipe(title=test_data[0], author_id=test_data[1], description=test_data[2],
```
We have only 1 recipe here

File: tasties_app/tests/test_Comment.py [hunk 18]
```diff
@@ -0,0 +1,47 @@
...
+                     directions=test_data[3], publication_date=timezone.now(), minutes_to_make=test_data[4],
```
Comment (liornoy) [inline tasties_app/tests/test_Comment.py:16]:
```suggestion
                     directions=test_data[3],  minutes_to_make=test_data[4],
```
We have this date as default already, so you can delete it.

File: tasties_app/tests/test_Comment.py [hunk 19]
```diff
@@ -0,0 +1,47 @@
...
+                     recipe_picture=test_data[5])
+    category1.save()
+    recipe1.save()
+    recipe1.categories.add(category1)
+    recipe1.save()
+    category1.save()
+    date1 = date.today().strftime("%Y-%m-%d")
+    comment = Comment(author_id=user3, recipe_id=recipe1, publication_date=date1, comment_text='test comment')
+    comment2 = Comment(author_id=user3, recipe_id=recipe1, publication_date=date1,
```
Comment (liornoy) [inline tasties_app/tests/test_Comment.py:25]:
```suggestion
    comment = Comment(author_id=user3, recipe_id=recipe1, comment_text='test comment')
    comment2 = Comment(author_id=user3, recipe_id=recipe1,
```
Same as above, we have defaults.

File: tasties_app/tests/test_Comment.py [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+
+
+class TestComment():
+    @pytest.mark.django_db
+    def test_str(self, generate_data):
+        user = User.objects.get(username='user3')
+        comment = Comment.objects.filter(author_id=user)
+        date1 = str(date.today().strftime("%Y-%m-%d"))
+        string1 = user.get_username()+' '+comment[0].comment_text+' '+date1
+        string2 = user.get_username()+' '+comment[1].comment_text+' '+date1
```
Comment (liornoy) [inline tasties_app/tests/test_Comment.py:38]: Please give string1, string2 variables meaningful names like: "expected_comment1_str"

File: tasties_app/tests/test_Comment.py [hunk 21]
```diff
@@ -0,0 +1,47 @@
...
+        assert string1 == comment[0].__str__()
+        assert string2 == comment[1].__str__()
+
+    @pytest.mark.django_db
+    def test_show_all_comment_author(self, generate_data):
+        user = User.objects.get(username='user3')
+        query1 = Comment.show_all_comment_author(user)
+        query2 = Comment.objects.filter(author_id=user)
+        assert (query2[0] and query2[1]) in query1
```
Comment (liornoy) [inline tasties_app/tests/test_Comment.py:47]: Let's remove that as well.

File: tasties_app/tests/test_Comment.py [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+from tasties_app.models import Comment, Recipe, Category
+from django.contrib.auth.models import User
+from django.utils import timezone
+from datetime import date
+
+
+@pytest.fixture
+def generate_data():
+    """This fixture create the data we need for tests
+    and save it in the data base. """
```
Comment (liornoy) [inline tasties_app/tests/test_Comment.py:11]: Please describe here what objects does it create in the database

File: tasties_app/tests/test_Comment.py [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+    comment.save()
+    comment2.save()
+
+
+class TestComment():
+    @pytest.mark.django_db
+    def test_str(self, generate_data):
+        user = User.objects.get(username='user3')
+        comment = Comment.objects.filter(author_id=user)
+        date1 = str(date.today().strftime("%Y-%m-%d"))
```
Comment (liornoy) [inline tasties_app/tests/test_Comment.py:36]:
```suggestion
        date = str(date.today().strftime("%Y-%m-%d"))
```
only 1 date. not need more numbering

**Review by liornoy — COMMENTED** [review]
A couple of comments. looks better!

**Comment by liornoy** [comment]
> > 
> 
> @liornoy I make changes please check. also i don't understand what you mean 'Command'?

I was talking about the commit message. it had a typo - Command instead of Comment.

File: tasties_app/models.py [hunk 24]
```diff
@@ -1,58 +1,57 @@
...
+class Ingredient(models.Model):
     recipe_id = models.ForeignKey(Recipe, on_delete=models.CASCADE)
-    amount = models.DecimalField(
-        max_digits=10, decimal_places=2, validators=[MinValueValidator(0.0001)]
-    )
-    measurement_unit = models.CharField(
-        max_length=11, choices=UnitChoices.choices, default=UnitChoices.WHOLE
-    )
+    amount = models.DecimalField(max_digits=10, decimal_places=2, validators=[MinValueValidator(0.0001)])
+    measurement_unit = models.TextChoices('measurement_unit', 'ML FLOZ TSP OZ CUP GRAM WHOLE')
```
Comment (liornoy) [inline tasties_app/models.py:30]: Why did you remove the class UnitChoices and switched to this?

**Review by efraimvis — CHANGES_REQUESTED** [review]
@WaseemAbuSobe Waseem, I think you should consider closing this PR, syncing your fork with the main repo, and applying your changes carefully to the most up to date version. This PR has been open for a long time, and in that time the main branch diverged significantly from what we have here, and we might have serious conflicts when trying to finally merge this PR. A fresh version of this PR, taking into consideration everything we've learned in the meanwhile (which is a lot) - would in my opinion be the best way forward.

**Comment by WaseemAbuSobe** [comment]
my main branch is all ready update.

File: tasties_app/tests/test_Comment.py [hunk 25]
```diff
@@ -0,0 +1,71 @@
...
+    test_data = ['Test Recipe1', user3, 'Test Description1', 'Test Directions1', 1, 'recipe.jpeg']
+    category1 = Category.objects.create(category_name="1")
+    recipe1 = Recipe(title=test_data[0], author_id=test_data[1], description=test_data[2],
+                     directions=test_data[3], publication_date=timezone.now(), minutes_to_make=test_data[4],
+                     recipe_picture=test_data[5])
+    category1.save()
+    recipe1.save()
+    recipe1.categories.add(category1)
+    recipe1.save()
+    category1.save()
```
[the remark this answers is not in the document]
Comment (WaseemAbuSobe) [reply tasties_app/tests/test_Comment.py:22]: @liornoy 
the Pytest does not work if i Delete this lines we need to save to fill my field Class.
if i delete this lines as also i need to delete line 14 and 15 we consider is not exist.

File: tasties_app/models.py (hunk 24 above)
Comment (liornoy) [reply tasties_app/models.py:30]: @WaseemAbuSobe did you mark this as resolved by mistake?
I still don't understand.

File: tasties_app/tests/test_Comment.py (hunk 16 above)
Comment (liornoy) [reply tasties_app/tests/test_Comment.py:14]: @WaseemAbuSobe you missed this comment.

File: tasties_app/tests/test_Comment.py (hunk 25 above)
[the remark this answers is not in the document]
Comment (liornoy) [reply tasties_app/tests/test_Comment.py:22]: I think we can delete line 14 and have a recipe without a category (correct me if I'm wrong)
And I meant just save the recipe object one time in the end.

File: tasties_app/models.py (hunk 24 above)
Comment (WaseemAbuSobe) [reply tasties_app/models.py]: @liornoy 
sorry. is was mistake now I fix.

**Comment by natifridman** [comment]
I think you should rebase with main, this branch doesn't look updated...

**Comment by WaseemAbuSobe** [comment]
> I think you should rebase with main, this branch doesn't look updated...
@natifridman 
I think now update ?

**Review by GalOved — CHANGES_REQUESTED** [review]
I think you need to add test that test if the user give a text less the minimum.
It is meaning the that the test need to check if someone edit the comment with a empty string.

**Review by GalOved — CHANGES_REQUESTED** [review]
Please add a test that checks if there is an update of an empty string.

**Review by natifridman — COMMENTED** [review]
Maybe you can put the recipes in a `media` folder, just in case you'll need to upload more files in the future.
So the recipe pictures will be under `media/recipe_pictures/`

**Comment by natifridman** [comment]
@WaseemAbuSobe Any update on this PR?

**Comment by WaseemAbuSobe** [comment]
> @WaseemAbuSobe Any update on this PR?

You can now check?

**Comment by natifridman** [comment]
> You can now check?

Per our [contribution guide](https://github.com/redhat-beyond/tasties/blob/main/CONTRIBUTING.md#pull-requests-standards) - `Each PR requires the approval of at least two team members`

**Review by GalOved — DISMISSED** [review]
LGTM

**Review by GilShechter — DISMISSED** [review]

File: .gitignore [hunk 26]
```diff
@@ -133,3 +133,5 @@ dmypy.json
 
 #Pycharm idea folder
 .idea
+
+.DS_Store
```
Comment (natifridman) [inline .gitignore:137]: Missing newline at end of file

**Review by natifridman — CHANGES_REQUESTED** [review]
Why do you need to use `gitkeep` in the `reciped_pictures` folder?
Also, for some reason, there are files not relevant for this PR here like the category migrations and all the tests files

**Comment by WaseemAbuSobe** [comment]
> Why do you need to use `gitkeep` in the `reciped_pictures` folder? Also, for some reason, there are files not relevant for this PR here like the category migrations and all the tests files
I don't can push empty folder so find this way in google by used  `.gitkeep`.
category migrations deleted .

**Comment by natifridman** [comment]
> I don't can push empty folder so find this way in google by used  `.gitkeep`.
> category migrations deleted .

There are still test files in this PR that shouldn't be here

**Comment by WaseemAbuSobe** [comment]
> > I don't can push empty folder so find this way in google by used  `.gitkeep`.
> > category migrations deleted .
> 
> There are still test files in this PR that shouldn't be here

what you mean `test files` ?

File: tasties_app/tests/test_Comment.py [hunk 27]
```diff
@@ -0,0 +1,42 @@
...
+@pytest.fixture
+def generate_comments():
+    """
+    in this generate function we create User, Category and Recipe
+    objects to fill Comment class and used to test
+     """
+    user = User.objects.create_user(username='test_user', password='password')
+    recipe = Recipe(title='Test Recipe1', author_id=user, description='Test Description1',
+                    directions='Test Directions1', publication_date=timezone.now(), minutes_to_make=1,
+                    recipe_picture='recipe.jpeg')
```
Comment (natifridman) [inline tasties_app/tests/test_Comment.py:17]: Where is this file?

**Review by natifridman — COMMENTED** [review]
@WaseemAbuSobe 

- tasties_app/tests/test_category.py
- tasties_app/tests/test_ingredient.py
- tasties_app/tests/test_recipe.py
All of these files shouldn't be part of this PR.

File: tasties_app/tests/test_Comment.py (hunk 27 above)
Comment (WaseemAbuSobe) [reply tasties_app/tests/test_Comment.py:17]: there is not need to the file its just for fill the field.

**Comment by WaseemAbuSobe** [comment]
> 

but I work according to the last update (make rebase)

**Comment by liornoy** [comment]
> > 
> 
> but I work according to the last update (make rebase)

image

If you enter the `git log` command on your branch, you can see it's not aligned with the last update.

**Comment by WaseemAbuSobe** [comment]
> > > 
> > 
> > 
> > but I work according to the last update (make rebase)
> 
> image
> 
> If you enter the `git log` command on your branch, you can see it's not aligned with the last update.

but according to my files it's the same with the main but the commits because i make squash to the same commit

**Comment by WaseemAbuSobe** [comment]
problem with the update
